test/integration/middleware_integration_test.rb

```diff
@@ -0,0 +1,26 @@
+require 'test_helper'
+
+describe 'Lotus middleware integration' do
+  before do
+    @routes = Lotus::Router.new(namespace: Web::Controllers) do
+      get '/',          to: 'home#index'
+      get '/dashboard', to: 'dashboard#index'
+    end
+
+    @app = Rack::MockRequest.new(@routes)
+  end
+
+  it 'action with middleware' do
+    response = @app.get('/')
+    response.body.must_equal 'Hello from Web::Controllers::Home::Index'
+    response.status.must_equal 200
+    response.headers.fetch('X-Middleware').must_equal 'CALLED'
+  end
+
+  it 'action without middleware' do
+    response = @app.get('/dashboard')
+    response.body.must_equal 'Hello from Web::Controllers::Dashboard::Index'
+    response.status.must_equal 200
+    -> { response.headers.fetch('X-Middleware') }.must_raise KeyError
```

Comment:
There is no need to raise an exception here, because we already know that the key is missing.
What if we change it with?

```ruby
response.headers['X-Middleware'].must_be_nil
```

:smile_cat: